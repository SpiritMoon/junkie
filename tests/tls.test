#!/bin/sh

NAME="TLS"
PCAPS="pcap/https/mycorp.pcap pcap/https/certificate.pcap pcap/https/tlsv1.pcap pcap/https/mycorp_with_sessions.pcap pcap/https/dokuwiki.pcap"
filter() {
	grep -e '^\(TLS\|HTTP\)'
}
OUTPUT="tls.out"
CMDLINE='
(tls-add-key 1 "pcap/https/mycorp.pem" "192.168.10.211" "255.255.255.255" #t "http")
(tls-add-key 2 "pcap/https/mycorp_with_sessions.pem" "192.168.10.9" "255.255.255.255" #t "http")
(tls-add-key 3 "pcap/https/dokuwiki.pem" "10.1.0.18" "255.255.255.255" #t "http")
'
# Add some more keys to shadow the correct one (for 10.1.0.18)
CMDLINE=$CMDLINE'
(tls-add-key 4 "pcap/https/mycorp_with_sessions.pem" "10.1.0.18" "255.255.255.255" #t "http")
(tls-add-key 5 "/should/not/exist" "10.1.0.18" "255.255.255.255" #t "http")
(tls-add-key 6 "tls.test" "10.1.0.18" "255.255.255.255" #t "http")
'

. "$srcdir/dump_test"
