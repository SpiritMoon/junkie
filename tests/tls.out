-- pcap/https/mycorp.pcap --
TLS: head_len=176, payload=0, version=3.1, content-type=handshake
TLS: head_len=58, payload=0, version=3.1, content-type=handshake, cipher_suite=RSA_WITH_AES_256_CBC_SHA, compression_algo=none
TLS: head_len=139, payload=0, version=3.1, content-type=handshake
TLS: head_len=207, payload=0, version=3.1, content-type=handshake
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=35, payload=210, version=3.1, content-type=data
HTTP: head_len=210, payload=0, method=unset, code=304, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.16 (Debian), orig_ip=unset, url=unset, pkts=1
TLS: head_len=37, payload=0, version=3.1, content-type=alert
TLS: head_len=37, payload=0, version=3.1, content-type=alert
-- pcap/https/certificate.pcap --
TLS: head_len=163, payload=0, version=3.1, content-type=handshake, SNI=github.com
TLS: head_len=58, payload=0, version=3.1, content-type=handshake, cipher_suite=RSA_WITH_RC4_128_SHA, compression_algo=none
TLS: head_len=3623, payload=0, version=3.1, content-type=handshake, nb_certs=2, serial_number_0=0x0e77768a5d07f0e57959ca2a9d5082b5, issuer_0=DigiCert High Assurance EV CA-1, subject_0=github.com, not_before_0=2011-05-27 00:00:00, not_after_0=2013-07-29 12:00:00, serial_number_1=0x08bbb02547134bc9b110d7c1a21259c5, issuer_1=DigiCert High Assurance EV Root CA, subject_1=DigiCert High Assurance EV CA-1, not_before_1=2006-11-10 00:00:00, not_after_1=2021-11-10 00:00:00
TLS: head_len=267, payload=0, version=3.1, content-type=handshake
TLS: head_len=191, payload=0, version=3.1, content-type=handshake
TLS: head_len=1251, payload=0, version=3.1, content-type=data
TLS: head_len=11270, payload=0, version=3.1, content-type=data
-- pcap/https/tlsv1.pcap --
TLS: head_len=167, payload=0, version=3.1, content-type=handshake, SNI=www.google.com
TLS: head_len=70, payload=0, version=3.1, content-type=handshake, cipher_suite=Unknown cipher suite 0xc011, compression_algo=none
TLS: head_len=1630, payload=0, version=3.1, content-type=handshake, nb_certs=2, serial_number_0=0x4f9d96d966b0992b54c2957cb4157d4d, issuer_0=Thawte SGC CA, subject_0=www.google.com, not_before_0=2011-10-26 00:00:00, not_after_0=2013-09-30 23:59:59, serial_number_1=0x30000002, issuer_1=, subject_1=Thawte SGC CA, not_before_1=2004-05-13 00:00:00, not_after_1=2014-05-12 23:59:59
TLS: head_len=75, payload=0, version=3.1, content-type=handshake
TLS: head_len=179, payload=0, version=3.1, content-type=handshake
TLS: head_len=1138, payload=0, version=3.1, content-type=data
TLS: head_len=936, payload=0, version=3.1, content-type=data
-- pcap/https/mycorp_with_sessions.pcap --
TLS: head_len=104, payload=0, version=3.1, content-type=handshake
TLS: head_len=86, payload=0, version=3.1, content-type=handshake, cipher_suite=RSA_WITH_AES_128_CBC_SHA, compression_algo=none
TLS: head_len=267, payload=0, version=3.1, content-type=handshake
TLS: head_len=6, payload=0, version=3.1, content-type=change_cipher_spec
TLS: head_len=37, payload=0, version=3.1, content-type=alert
TLS: head_len=136, payload=0, version=3.1, content-type=handshake
TLS: head_len=86, payload=0, version=3.1, content-type=handshake, cipher_suite=RSA_WITH_AES_128_CBC_SHA, compression_algo=none
TLS: head_len=6, payload=0, version=3.1, content-type=change_cipher_spec
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=26, payload=251, version=3.1, content-type=data
HTTP: head_len=251, payload=0, method=unset, code=200, content_length=816, transfert_encoding=unset, mime_type=text/html;charset=UTF-8, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Debian), orig_ip=unset, url=unset, pkts=1, compressed
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=35, payload=290, version=3.1, content-type=data
HTTP: head_len=290, payload=0, method=unset, code=200, content_length=148, transfert_encoding=unset, mime_type=image/gif, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Debian), orig_ip=unset, url=unset, pkts=1
TLS: head_len=37, payload=0, version=3.1, content-type=alert
-- pcap/https/dokuwiki.pcap --
TLS: head_len=165, payload=0, version=3.1, content-type=handshake, SNI=dokuwiki.securalis.lan
TLS: head_len=62, payload=0, version=3.1, content-type=handshake, cipher_suite=RSA_WITH_AES_256_CBC_SHA, compression_algo=none
TLS: head_len=139, payload=0, version=3.1, content-type=handshake
TLS: head_len=223, payload=0, version=3.1, content-type=handshake
TLS: head_len=37, payload=0, version=3.1, content-type=alert
TLS: head_len=405, payload=0, version=3.1, content-type=handshake, SNI=dokuwiki.securalis.lan
TLS: head_len=86, payload=0, version=3.1, content-type=handshake, cipher_suite=RSA_WITH_AES_256_CBC_SHA, compression_algo=none
TLS: head_len=6, payload=0, version=3.1, content-type=change_cipher_spec
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=34, payload=627, version=3.1, content-type=data
HTTP: head_len=627, payload=0, method=unset, code=200, content_length=unset, transfert_encoding=chunked, mime_type=text/html; charset=utf-8, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1, compressed
TLS: head_len=32, payload=1813, version=3.1, content-type=data
HTTP: head_len=0, payload=1813, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=4
TLS: head_len=33, payload=4, version=3.1, content-type=data
TLS: head_len=32, payload=5, version=3.1, content-type=data
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=37, payload=432, version=3.1, content-type=data
HTTP: head_len=432, payload=0, method=unset, code=200, content_length=17068, transfert_encoding=unset, mime_type=text/css; charset=utf-8, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1, compressed
TLS: head_len=37, payload=8096, version=3.1, content-type=data
HTTP: head_len=0, payload=8096, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=3
TLS: head_len=37, payload=8096, version=3.1, content-type=data
HTTP: head_len=0, payload=8096, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=4
TLS: head_len=27, payload=858, version=3.1, content-type=data
HTTP: head_len=0, payload=858, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=5
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=34, payload=291, version=3.1, content-type=data
HTTP: head_len=291, payload=0, method=unset, code=200, content_length=1541, transfert_encoding=unset, mime_type=image/png, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=32, payload=1541, version=3.1, content-type=data
HTTP: head_len=0, payload=1541, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=2
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=35, payload=290, version=3.1, content-type=data
HTTP: head_len=290, payload=0, method=unset, code=200, content_length=316, transfert_encoding=unset, mime_type=image/png, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=33, payload=292, version=3.1, content-type=data
HTTP: head_len=292, payload=0, method=unset, code=200, content_length=5931, transfert_encoding=unset, mime_type=image/png, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=26, payload=5931, version=3.1, content-type=data
HTTP: head_len=0, payload=5931, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=2
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=35, payload=290, version=3.1, content-type=data
HTTP: head_len=290, payload=0, method=unset, code=200, content_length=816, transfert_encoding=unset, mime_type=image/png, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=26, payload=267, version=3.1, content-type=data
HTTP: head_len=267, payload=0, method=unset, code=200, content_length=541, transfert_encoding=unset, mime_type=application/opensearchdescription+xml; charset=utf-8, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=30, payload=551, version=3.1, content-type=data
HTTP: head_len=551, payload=0, method=unset, code=200, content_length=unset, transfert_encoding=chunked, mime_type=text/html; charset=utf-8, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1, compressed
TLS: head_len=41, payload=1836, version=3.1, content-type=data
HTTP: head_len=0, payload=1836, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=4
TLS: head_len=32, payload=5, version=3.1, content-type=data
TLS: head_len=29, payload=1976, version=3.1, content-type=data
HTTP: head_len=0, payload=1976, method=unset, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=7
TLS: head_len=32, payload=5, version=3.1, content-type=data
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=36, payload=289, version=3.1, content-type=data
HTTP: head_len=289, payload=0, method=unset, code=200, content_length=121, transfert_encoding=unset, mime_type=image/png, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=36, payload=1, version=3.1, content-type=data
TLS: head_len=36, payload=289, version=3.1, content-type=data
HTTP: head_len=289, payload=0, method=unset, code=200, content_length=225, transfert_encoding=unset, mime_type=image/png, host=unset, user_agent=unset, referrer=unset, server=Apache/2.2.22 (Ubuntu), orig_ip=unset, url=unset, pkts=1
TLS: head_len=37, payload=0, version=3.1, content-type=alert
